Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs

```diff
@@ -0,0 +1,313 @@
+
+                state.numFocusChanged++;
+            }
+
+            void IMixedRealityFocusHandler.OnFocusEnter(FocusEventData eventData)
+            {
+                Assert.AreEqual(1, numFocusing);
+
+                state.numFocusEnter++;
+            }
+
+            void IMixedRealityFocusHandler.OnFocusExit(FocusEventData eventData)
+            {
+                Assert.AreEqual(0, numFocusing);
+
+                state.numFocusExit++;
+            }
+
+            void IMixedRealityPointerHandler.OnPointerClicked(MixedRealityPointerEventData eventData)
+            {
+                state.numPointerClicked++;
+            }
+
+            void IMixedRealityPointerHandler.OnPointerDown(MixedRealityPointerEventData eventData)
+            {
+                state.numPointerDown++;
+            }
+
+            void IMixedRealityPointerHandler.OnPointerDragged(MixedRealityPointerEventData eventData)
+            {
+                state.numPointerDragged++;
+            }
+
+            void IMixedRealityPointerHandler.OnPointerUp(MixedRealityPointerEventData eventData)
+            {
+                state.numPointerUp++;
+            }
+        }
+
+        // Utility class to use a simulated hand
+        private class Hand
+        {
+            private Handedness handedness;
+            private Vector3 position;
+            private ArticulatedHandPose.GestureId gestureId = ArticulatedHandPose.GestureId.Open;
+            private InputSimulationService simulationService;
+
+            public Hand(Handedness handedness)
+            {
+                this.handedness = handedness;
+                simulationService = PlayModeTestUtilities.GetInputSimulationService();
+            }
+
+            public IEnumerator Show(Vector3 position)
+            {
+                this.position = position;
+                return PlayModeTestUtilities.ShowHand(handedness, simulationService, position);
+            }
+
+            public IEnumerator Hide()
+            {
+                return PlayModeTestUtilities.HideHand(handedness, simulationService);
+            }
+
+            public IEnumerator MoveTo(Vector3 newPosition, int numSteps = 30)
+            {
+                Vector3 oldPosition = position;
+                position = newPosition;
+                return PlayModeTestUtilities.MoveHandFromTo(oldPosition, newPosition, numSteps, gestureId, handedness, simulationService);
+            }
+
+            public IEnumerator SetGesture(ArticulatedHandPose.GestureId newGestureId)
+            {
+                gestureId = newGestureId;
+                return PlayModeTestUtilities.MoveHandFromTo(position, position, 1, gestureId, handedness, simulationService);
+            }
+        }
+
+        // Initializes MRTK, instantiates the test content prefab and adds a pointer handler to the test collider
+        private PointerHandler SetupScene()
+        {
+            TestUtilities.InitializeMixedRealityToolkit(true);
```

Comment:
Have a look at how lars is setting up and destroying tests in #5033. I think Lars will soon get his changes in, you'll probably need to update setup/teardown to match that pattern.

Reply:
We should probably do a batch update of all tests when Lars's changes are in.